### PR TITLE
Make neo4j 5.X driver compatible

### PR DESCRIPTION
**Pull request by adamnsch**
Co-Authored-By: Florentin Dörre <florentin.dorre@neotechnology.com>
Co-Authored-By: Brian Shi <brian.shi@neotechnology.com>

Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (knutwalker) [inline setup.py:23]: Do we need to note something in the docs or the readme, that 3.10 is only supported when the neo4j 5.x driver is used? Or are we supporting 3.10 for all versions?

**Review by knutwalker — APPROVED** [review]
👍

File: setup.py (hunk 1 above)
Comment (adamnsch) [reply setup.py:23]: Good points. We need to add a new badge to the readme, and to the compat matrix in the docs. I think pip will handle the fact that 5.X should only be used with py3.10

**Comment by adamnsch** [comment]
> +1

@knutwalker @FlorentinD  To make the tests pass, which of the following would you prefer?

1. Catch the deprecation warning for now
2. Run the encrypted tests without `-Werror`

**Comment by FlorentinD** [comment]
I would opt for caching it for now

File: setup.py (hunk 1 above)
Comment (adamnsch) [reply setup.py:23]: done

**Pull request merged**